Editor/EditorCore/HierarchyListener.cs

```diff
@@ -45,11 +45,11 @@ static void ObjectEventChangesPublished(ref ObjectChangeEventStream stream)
                     stream.GetChangeGameObjectStructureEvent(i, out var data);
                     GameObjectCreatedOrStructureModified(data.instanceId);
                 }
-                #if !UNITY_2023_1_OR_NEWER
-                // for handling prefab revert pre-2023.1
                 else if (stream.GetEventType(i) == ObjectChangeKind.ChangeGameObjectStructureHierarchy)
                 {
-                    // note that this is leaking meshes when reverting! in 2023.1+ we handle it correctly, but 2022 and
+                    // Note 2 : This needs to be called when using a Prefab>Replace action in the menus to refresh the current
```

Comment:
Is this the correct event to handle prefab replace? If I remember correctly, `ChangeGameObjectStructureHierarchy` is one of the more frequently called events. 

 I'm surprised that this isn't covered by `PrefabInstanceUpdated`, for instance.